lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +1004,59 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        if module.check_mode:
+            result.update(changed=True)
+            module.exit_json(**result)
+
+        all_archive_fmt = {'.zip': 'zip', '.gz': 'tar.gz', '.tar': 'tar',
+                           '.tgz': 'tgz'}
+        _, archive_ext = os.path.splitext(archive)
+        archive_fmt = all_archive_fmt.get(archive_ext, None)
+        if archive_fmt is None:
+            module.fail_json(msg="Unable to get file extension from "
+                                 "archive file name : %s" % archive,
+                             details="Please specify archive as filename with "
+                                     "extension. File extension can be one "
+                                     "of ['tar', 'tar.gz', 'zip', 'tgz']")
+
+        repo_name = repo.split("/")[-1].replace(".git", "")
+        # Git archive is not supported by all git servers, so
+        # we will first clone and perform git archive from local directory
+        if dest and not os.path.exists(gitconfig):
+            clone(git_path, module, repo, dest, remote, depth, version, bare,
```

Comment:
@robinro Do you mean after [this](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L1039) code ? I added archive code here because I didn't want to complicate other part of code.